modules/RTC/RTC.js

```diff
@@ -260,94 +362,134 @@ export default class RTC extends Listenable {
      * @returns {Promise}
      */
     setAudioMute (value) {
-        var mutePromises = [];
-        for(var i = 0; i < this.localTracks.length; i++) {
-            var track = this.localTracks[i];
-            if(track.getType() !== MediaType.AUDIO) {
-                continue;
-            }
+        const mutePromises = [];
+        this.getLocalTracks(MediaType.AUDIO).forEach(function(audioTrack){
             // this is a Promise
-            mutePromises.push(value ? track.mute() : track.unmute());
-        }
+            mutePromises.push(value ? audioTrack.mute() : audioTrack.unmute());
+        });
         // we return a Promise from all Promises so we can wait for their execution
         return Promise.all(mutePromises);
     }
 
     removeLocalTrack (track) {
-        var pos = this.localTracks.indexOf(track);
+        const pos = this.localTracks.indexOf(track);
         if (pos === -1) {
             return;
         }
 
         this.localTracks.splice(pos, 1);
-
-        if (track.isAudioTrack()) {
-            this.localAudio = null;
-        } else {
-            this.localVideo = null;
-        }
     }
 
     /**
-     * Initializes a new JitsiRemoteTrack instance with the data provided by (a)
-     * ChatRoom to XMPPEvents.REMOTE_TRACK_ADDED.
+     * Initializes a new JitsiRemoteTrack instance with the data provided by
+     * the signalling layer and SDP.
      *
-     * @param {Object} event the data provided by (a) ChatRoom to
-     * XMPPEvents.REMOTE_TRACK_ADDED to (a)
-     */
-    createRemoteTrack (event) {
-        var ownerJid = event.owner;
-        var remoteTrack = new JitsiRemoteTrack(
-            this, this.conference, ownerJid, event.stream, event.track,
-            event.mediaType, event.videoType, event.ssrc, event.muted);
-        var resource = Strophe.getResourceFromJid(ownerJid);
-        var remoteTracks
-            = this.remoteTracks[resource] || (this.remoteTracks[resource] = {});
-        var mediaType = remoteTrack.getType();
+     * @param {string} owner
```

Comment:
what is the owner here? maybe use a more descriptive name as suggested above by @bbaldino ?